src/components/link/link.test.js

```diff
@@ -0,0 +1,178 @@
+import * as React from "react";
+import Link from "./link.component";
+
+import {
+  icon,
+  tooltipPreview,
+  link,
+  getDataElementByValue,
+  body,
+} from "../../../cypress/locators";
+import { skipLink } from "../../../cypress/locators/link/index";
+import { keyCode } from "../../../cypress/support/helper";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+const testCypress = "test-cypress";
+
+const LinkComponent = ({ ...props }) => {
+  return (
+    <div
+      style={{
+        margin: "80px",
+      }}
+    >
+      <Link href="#foo" target="_blank" rel="noreferrer noopener" {...props}>
+        This is a link
+      </Link>
+    </div>
+  );
+};
+
+context("Test for Link component", () => {
+  describe.only("Check props for Link component", () => {
+    it.each(testData)(
+      "should render Link className using %s special characters",
+      (className) => {
+        CypressMountWithProviders(<LinkComponent className={className} />);
+
+        link().should("have.attr", "class").and("contain", className);
+      }
+    );
+
+    it("should render Link disabled", () => {
+      CypressMountWithProviders(<LinkComponent disabled />);
+
+      link().should("have.attr", "disabled");
+      link().children().should("have.css", "color", "rgba(0, 0, 0, 0.3)");
```

Comment:
Should we care about the background color here? Wouldn't "link().should("have.attr", "disabled");" be enough to test the "disabled" prop?

Reply:
Actually the color is changing for the `children`, so I think so :)